src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/TypeInformationIT.java

```diff
@@ -0,0 +1,200 @@
+        JSONObject response = executeJdbcRequest("SELECT ABS(age) FROM " + TestsConstants.TEST_INDEX_ACCOUNT +
+                " ORDER BY age LIMIT 5");
+
+        verifySchema(response, schema("ABS(age)", null, "long"));
+        verifyDataRows(response,
+                rows(20),
+                rows(20),
+                rows(20),
+                rows(20),
+                rows(20));
+    }
+
+    @Test
+    public void testCeilWithLongFieldReturnsLong() {
+        JSONObject response = executeJdbcRequest("SELECT CEIL(balance) FROM " + TestsConstants.TEST_INDEX_ACCOUNT +
+                " ORDER BY balance LIMIT 5");
+
+        verifySchema(response, schema("CEIL(balance)", null, "long"));
+        verifyDataRows(response,
+                rows(1011),
+                rows(1031),
+                rows(1110),
+                rows(1133),
+                rows(1172));
+    }
+
+    /*
+    mathConstants
+     */
+    @Test
+    public void testPiReturnsDouble() {
+        JSONObject response = executeJdbcRequest("SELECT PI() FROM " + TestsConstants.TEST_INDEX_ACCOUNT
+                + " LIMIT 1");
+
+        verifySchema(response, schema("PI()", null, "double"));
+        verifyDataRows(response,
+                rows(3.141592653589793));
+    }
+
+    /*
+    stringOperators
+     */
+    @Test
+    public void testUpperWithStringFieldReturnsString() {
+        JSONObject response = executeJdbcRequest("SELECT UPPER(firstname) AS firstname_alias FROM " +
+                TestsConstants.TEST_INDEX_ACCOUNT + " ORDER BY firstname_alias LIMIT 2");
+
+        verifySchema(response, schema("firstname_alias", null, "text"));
+        verifyDataRows(response,
+                rows("ABBOTT"),
+                rows("ABIGAIL"));
+    }
+
+    @Test
+    public void testLowerWithTextFieldReturnsText() {
+        JSONObject response = executeJdbcRequest("SELECT LOWER(firstname) FROM " +
+                TestsConstants.TEST_INDEX_ACCOUNT + " ORDER BY firstname LIMIT 2");
+
+        verifySchema(response, schema("LOWER(firstname)", null, "text"));
+        verifyDataRows(response,
+                rows("abbott"),
+                rows("abigail"));
+    }
+
+    /*
+    stringFunctions
+     */
+    @Test
+    public void testLengthWithTextFieldReturnsInt() {
+        JSONObject response = executeJdbcRequest("SELECT length(firstname) FROM " +
+                TestsConstants.TEST_INDEX_ACCOUNT + " ORDER BY firstname LIMIT 2");
+
+        verifySchema(response, schema("length(firstname)", null, "integer"));
+        verifyDataRows(response,
+                rows(6),
+                rows(7));
+    }
+
+    /*
+    trigFunctions
+     */
+    @Test
+    public void testSinWithLongFieldReturnsLong() {
+        JSONObject response = executeJdbcRequest("SELECT sin(balance) FROM " +
+                TestsConstants.TEST_INDEX_ACCOUNT + " ORDER BY firstname LIMIT 2");
+
+        verifySchema(response, schema("sin(balance)", null, "long"));
```

Comment:
I think I made this mistake. `sin` should always return double like `log` right? Could you help confirm and fix this as well as `TAN`, `ATAN`, `ATAN2`, `COS` and `COSH` in `ScalarFunction` type system?

For example, change `SIN(func(T(NUMBER)).to(T)),` to `SIN(func(T(NUMBER)).to(DOUBLE)),`.

Reply:
Sure- will look into it

Reply:
Thanks!